src/jshint.js

```diff
@@ -1419,12 +1425,23 @@ var JSHINT = (function() {
     var x = symbol(s, 150);
 
     x.led = function(left) {
+      // this = suffix e.g. "++" punctuator
+      // left = symbol operated e.g. "a" identifier or "a.b" punctuator
+
       if (state.option.plusplus) {
         warning("W016", this, this.id);
       } else if ((!left.identifier || isReserved(left)) && left.id !== "." && left.id !== "[") {
         warning("W017", this);
       }
 
+      // detect modification of a const.
+      // in the case of a.b, left will be the "." punctuator
```

Comment:
You deleted the comment on github (or it otherwise vanished for some other reason), so I wasn’t able to reply on github and can’t see what you’re talking about. Anyways, the point is pretty clear: always document the intentions of your changes. The pratt parser is a mess, and not documenting changes is a real problem for maintainability. Just make sure you’re always crystal clear about the inputs you’re expecting and why

> On Apr 26, 2015, at 11:36 AM, Luke Page <notifications@github.com> wrote:
> 
> In src/jshint.js <https://github.com/jshint/jshint/pull/2326#discussion_r29110918>:
> 
> >        if (state.option.plusplus) {
> >          warning("W016", this, this.id);
> >        } else if ((!left.identifier || isReserved(left)) && left.id !== "." && left.id !== "[") {
> >          warning("W017", this);
> >        }
> >  
> > +      // detect modification of a const.
> > +      // in the case of a.b, left will be the "." punctuator
> It’s called for a number of prefixes, preinc, predec, unary minus, unary plus. The unary ops are totally fine because they’re not assignment ops.
> 
> you put this comment on the postfix function, which is only called for ++ and --
> 
> The prefix function this is the case, but in that function my block is clearly in the if(this.id === "--" || this.id === "++") section. anyway, as I said I've changed the comment anyway in both places, is that what you want?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/jshint/jshint/pull/2326/files#r29110918>.
>